e2e/recipes/recipes_test.go

```diff
@@ -0,0 +1,364 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+//nolint:testpackage
+package recipes_test
+
+import (
+	"context"
+	"testing"
+
+	recipe "github.com/ramendr/recipe/api/v1alpha1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/ramendr/ramen/e2e/app"
+	"github.com/ramendr/ramen/e2e/config"
+	"github.com/ramendr/ramen/e2e/deployers"
+	"github.com/ramendr/ramen/e2e/helpers"
+	"github.com/ramendr/ramen/e2e/recipes"
+	"github.com/ramendr/ramen/e2e/test"
+	"github.com/ramendr/ramen/e2e/types"
+	"github.com/ramendr/ramen/e2e/workloads"
+)
+
+var group = map[string]string{"group": "rg1"}
+
+func TestGenerateWithNoHooks(t *testing.T) {
+	recipeConfig := &config.Recipe{
+		Type:      "generate",
+		CheckHook: false,
+		ExecHook:  false,
+	}
+	testContext := createTestContext(t, recipeConfig)
+
+	actualRecipe := recipes.Generate(testContext, recipeConfig)
+
+	expectedRecipe := &recipe.Recipe{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       recipes.KindRecipe,
+			APIVersion: recipes.RecipeAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testContext.Name(),
+			Namespace: testContext.AppNamespace(),
+		},
+		Spec: recipe.RecipeSpec{
+			Groups: []*recipe.Group{
+				{
+					Name:      recipes.DefaultGroupName,
+					Type:      recipes.Resource,
+					BackupRef: recipes.DefaultGroupName,
+					IncludedNamespaces: []string{
+						testContext.AppNamespace(),
+					},
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"appname": "busybox",
+						},
+					},
+				},
+			},
+			Workflows: []*recipe.Workflow{
+				{
+					Name:     recipes.Backup,
+					Sequence: []map[string]string{group},
```

Comment:
Will be much easier to understand as:

    Sequence: []map[string]string{{"group": "rg1"}},